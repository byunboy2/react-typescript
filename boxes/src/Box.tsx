/** Colored box presentation, along with removeButton
 *
 * Props:
 * - id (unique)
 * - width
 * - height
 * - backgroundColor
 * - remove (function to call)
 *
 * BoxList -> Box
 */

interface BoxProps {
  id: string;
  width: number;
  height: number ;
  backgroundColor: string;
  remove: Function;
}



function Box({ id, width = 5, height = 5, backgroundColor, remove }: BoxProps) {

  /** Remove a box. */
  function handleRemove() {
    remove(id);
  }

  return (
    <div className="Box">
      <div className="Box-box"
        style={{
          height: `${height}em`,
          width: `${width}em`,
          backgroundColor: backgroundColor
        }}
      />
      <button
        className="Box-removeBtn"
        onClick={handleRemove}>
        Remove The Box!
      </button>
    </div>
  );
}

export default Box;
