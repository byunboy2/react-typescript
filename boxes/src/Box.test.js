import { render } from "@testing-library/react";
import Box from "./Box"

describe("Tests the Box component", function () {

  it("renders without crashing", function () {
    render(<Box />);
  });

  it("matches the snapshot", function () {
    const { container } = render(<Box />);
    expect(container).toMatchSnapshot();
  });
});
