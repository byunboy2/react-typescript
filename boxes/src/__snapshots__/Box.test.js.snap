// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Box component matches the snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 5em; width: 5em;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;

exports[`matches the snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 5em; width: 5em;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
